src/System.Buffers.Primitives/System/Buffers/IOutput.cs

```diff
@@ -5,10 +5,9 @@ namespace System.Buffers
 {
     public interface IOutput
     {
-        Span<byte> GetSpan();
         void Advance(int bytes);
 
         /// <summary>desiredBufferLength == 0 means "i don't care"</summary>
-        void Enlarge(int desiredBufferLength = 0);
+        Memory<byte> GetMemory(int desiredBufferLength = 0);
```

Comment:
Why are we changing it from Span to Memory?

Reply:
Also, let's rename the parameter to minimumLength.

Reply:
Also, let's describe the contract in the doc comment.

Reply:
It was in the email:

1.	It aligns with ROB that has memory and Reader/Writer that operate on span
2.	Allows IOuput to be used with async methods (scenario: get memory from IOutput, pass to Stread.ReadAsync)